syntaxes/alex.YAML-tmLanguage

```diff
@@ -0,0 +1,103 @@
+fileTypes:
+      - x
+name: Haskell Alex
+patterns:
+  - include: "#comments"
+  - include: "#blocks"
+  - include: "#syntax"
+  - include: "#strings"
+repository:
+  block_comment:
+    applyEndPatternLast: 1
+    begin: "{-"
+    captures:
+      '0': {name: punctuation.definition.comment.alex}
+    end: "-}"
+    name: comment.block.alex
+    patterns:
+      - include: "#block_comment"
+  comments:
+    patterns:
+      - name: comment.line.alex
+        match: '\-\-.*$'
+      - include: "#block_comment"
+  blocks:
+    patterns:
+      - name: meta.block.startcode.alex
+        begin: (<)\s*(\w*)(?:\s*(,)\s*(\w*))*\s*(>)\s*(\{)
+        beginCaptures:
+          "0": { name: punctuation.bracket.startcode.alex }
+          "1": { name: variable.other.starcode.alex }
+          "3": { name: punctuation.comma.startcode.alex }
+          "4": { name: variable.other.startcode.alex }
+          "5": { name: punctuation.bracket.startcode.alex }
+          "6": { name: punctuation.block.startcode.begin.alex }
```

Comment:
Good call! Any idea why my test case passed ok? It seems like at least ` "1": { name: variable.other.starcode.alex }` should have caused my test case to fail...

Reply:
Yes, the tests that handle syntax near the beginning of lines are incorrect. For instance, you have:

```
<foo,bar> {
-- <       - punctuation.brackets.startcode.alex
```

That line is ignored entirely, because it doesn't pick out code to check against. The syntax for something that starts at the beginning is

```
noYes
<~~--- yes
```

so you include as many tildes as characters you want to ignore, and then as many hyphens as characters you want to match. Note that the `<` does not count in any way, only the hyphens.   
Also be careful that if you include a hyphen that is disconnected from the `<~~~--` or the `^^^^` then that is counted as a negation, i.e. "this piece of code should not have this scope".

Reply:
Perfect! this should be the last piece hopefully, I'll get this fixed